sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala

```diff
@@ -49,6 +49,24 @@ object CodeFormatter {
     }
     code.result()
   }
+
+  def stripOverlappingComments(codeAndComment: CodeAndComment): CodeAndComment = {
+    val code = new StringBuilder
+    val map = codeAndComment.comment
+    var lastLine: String = "dummy"
+    codeAndComment.body.split('\n').foreach { l =>
+      val line = l.trim()
+      val skip = lastLine.startsWith("/*") && lastLine.endsWith("*/") &&
+        line.startsWith("/*") && line.endsWith("*/") &&
+        map(lastLine).substring(3).contains(map(line).substring(3))
```

Comment:
I think it's okay for the performance.
- This function is used for at every `CodeAndComment` creation once.
- It scans `codeAndComment.body` once.
- Map lookup occurs on each line at most once. Also, it does not cost much in this case.